Support for AllTalk V1 and V2

erew123 opened the pull request:
Select the version in the interface. 
I am providing an update to the AllTalk extension so that it works with both AllTalk V1 servers and AllTalk v2 servers. 

<!-- Put X in the box below to confirm -->

## Checklist:

- [x ] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

erew123 committed:
Support for AllTalk V1 and V2

Select the version in the interface. 
RVC is not supported on AllTalk V1

Cohee1207 commented on public/scripts/extensions/tts/alltalk.js:
```diff
@@ -1,840 +1,1113 @@
+        const languageSelect = document.getElementById('language_options');
+        if (languageSelect) {
+            // Ensure default language is set
+            this.settings.language = this.settings.language;
```
This line has no effect.

Cohee1207 commented on public/scripts/extensions/tts/alltalk.js:
```diff
@@ -1,840 +1,1113 @@
+            });
+        }
+
+        function lowvramdebounce(func, delay) {
```
We have a shared debounce utility function that you can import from scripts/utils.js

Cohee1207 reviewed (changes requested):
1. Please provide a cleaner diff. This is hard to review without external tools, this is required to make sure that we're not losing any bugfixes made in the code base after the initial commit.
2. Please run eslint (`npm run lint`) on the changes and fix the highlighted issues. The consistent code style is important for maintainability.

erew123 committed:
Updated AllTalk Extension to support AllTalk V2

1. Core Feature Addition:
   - Added AllTalk V1/V2 server version selection
   - Added RVC (Realistic Voice Conversion) support for V2
   - RVC features are automatically disabled when V1 is selected

2. Code Improvements:
   - Replaced custom debounce implementation with shared utils.js debounce utility
   - Fixed linting issues:
     - Converted HTML attribute quotes in template literals to single quotes
     - Added trailing commas where required
     - Fixed console.error message formatting

3. New Settings/Properties Added:
   ```javascript
   server_version: 'v2' (default)
   rvc_character_voice: 'Disabled'
   rvc_character_pitch: '0'
   rvc_narrator_voice: 'Disabled'
   rvc_narrator_pitch: '0'
   ```

4. Bug Fixes/Improvements:
   - Better error handling for RVC voice fetching
   - Improved URL handling for V1/V2 differences in API responses
   - Enhanced settings initialization and validation

5. Structural Changes:
   - Added RVC-specific UI elements and controls
   - Added version-specific logic for API endpoints
   - Improved settings synchronization between UI and backend

**NOTE** On line 70 there is an eslint bypass:

```javascript
        // HTML template literals can trigger ESLint quotes warnings when quotes are used in HTML attributes.
        // Disabling quotes rule for this one line as it's a false positive with HTML template literals.
        // eslint-disable-next-line quotes
        let html = `<div class="at-settings-separator">AllTalk V2 Settings</div>`;
```

The reason is:
1. ESLint's quotes rule wants all strings to use single quotes
2. However, this is a template literal containing HTML, where double quotes are standard for attributes
3. I tried various solutions:
   - Using single quotes: `<div class='at-settings-separator'>`
   - Using double quotes: `<div class="at-settings-separator">`
   - Even tried escaping quotes
   But ESLint just kept flagging it as an error

erew123 commented:
Hi @Cohee1207 

Hopefully, this should cover everything, but let me know if not!

Thanks

erew123

### Updates in the extension
1. Core Feature Addition:
   - Added AllTalk V1/V2 server version selection
   - Added RVC (Realistic Voice Conversion) support for V2
   - RVC features are automatically disabled when V1 is selected

2. Code Improvements:
   - Replaced custom debounce implementation with shared utils.js debounce utility
   - Fixed linting issues:
     - Converted HTML attribute quotes in template literals to single quotes
     - Added trailing commas where required
     - Fixed console.error message formatting

3. New Settings/Properties Added:
   ```javascript
   server_version: 'v2' (default)
   rvc_character_voice: 'Disabled'
   rvc_character_pitch: '0'
   rvc_narrator_voice: 'Disabled'
   rvc_narrator_pitch: '0'
   ```

4. Bug Fixes/Improvements:
   - Better error handling for RVC voice fetching
   - Improved URL handling for V1/V2 differences in API responses
   - Enhanced settings initialization and validation

5. Structural Changes:
   - Added RVC-specific UI elements and controls
   - Added version-specific logic for API endpoints
   - Improved settings synchronization between UI and backend

**NOTE** On line 70 there is an eslint bypass:

```javascript
        // HTML template literals can trigger ESLint quotes warnings when quotes are used in HTML attributes.
        // Disabling quotes rule for this one line as it's a false positive with HTML template literals.
        // eslint-disable-next-line quotes
        let html = `<div class="at-settings-separator">AllTalk V2 Settings</div>`;
```

The reason is:
1. ESLint's quotes rule wants all strings to use single quotes
2. However, this is a template literal containing HTML, where double quotes are standard for attributes
3. I tried various solutions:
   - Using single quotes: `<div class='at-settings-separator'>`
   - Using double quotes: `<div class="at-settings-separator">`
   - Even tried escaping quotes
   But ESLint just kept flagging it as an error

Cohee1207 reviewed (approved):
Looks good now, thanks. Will test and merge tonight.

Cohee1207 commented:
Working.

erew123 commented:
@Cohee1207 Awesome! Thanks so much :)
